tgui/packages/tgui/interfaces/DecalPainter.tsx

```diff
@@ -52,82 +52,88 @@ export const DecalPainter = (props) => {
+              })}
+              {supports_custom_color && (
+                <Button
+                  selected={custom_color_selected}
+                  onClick={() => act('pick custom color')}
                 >
-                  {data.dir_list.map((dir) => {
-                    const selected =
-                      decal.decal === data.current_decal &&
-                      dir.dir === data.current_dir;
-
-                    return (
-                      <IconButton
-                        key={dir.dir}
-                        decal={decal.decal}
-                        dir={dir.dir}
-                        color={preview_color}
-                        label={`${dir.name} ${decal.name}`}
-                        selected={selected}
-                      />
-                    );
-                  })}
-                </Flex>
-              );
-            })}
-          </Flex>
-        </Section>
+                  <ColorBox color={data.current_custom_color} mr={0.5} />
+                  Custom
+                </Button>
+              )}
+            </Section>
+          </Stack.Item>
+          <Stack.Item grow style={{ overflowY: 'auto', overflowX: 'hidden' }}>
+            <Section title="Decal Style" scrollable fill>
+              <Flex direction="row" wrap="wrap" align="fill" justify="fill">
+                {data.decal_list.map((decal) => {
+                  const nondirectional = data.nondirectional_decals.includes(
+                    decal.decal,
+                  );
+
+                  return nondirectional ? (
+                    // Tallll button for nondirectional
+                    <IconButton
+                      key={decal.decal}
+                      decal={decal.decal}
+                      dir={2}
+                      color={preview_color}
+                      label={decal.name}
+                      selected={decal.decal === data.current_decal}
+                    />
+                  ) : (
+                    // 4 buttons for directional
+                    <Flex
+                      key={decal.decal}
+                      direction="column"
+                      wrap="nowrap"
+                      align="fill"
+                      justify="fill"
+                    >
+                      {data.dir_list.map((dir) => {
+                        const selected =
+                          decal.decal === data.current_decal &&
+                          dir.dir === data.current_dir;
+
+                        return (
+                          <IconButton
+                            key={dir.dir}
+                            decal={decal.decal}
+                            dir={dir.dir}
+                            color={preview_color}
+                            label={`${dir.name} ${decal.name}`}
+                            selected={selected}
+                          />
+                        );
+                      })}
+                    </Flex>
```

Comment:
1. probably valid to break into its own component
2. more weird flex props. 
3. why not use stack?

Reply:
I'll be honest on this, no clue what any of that means. I didn't change any code in here, this is how the UI already worked.

I understand the flex thing though I can see if removing those breaks it

Reply:
you can (and should) abstract chunks of react code when it makes it more readable to do so - in this case, you can pull this entire block into its own subcomponent. the size of a react component should be 30-100 lines ish